mozci/push.py

```diff
@@ -341,7 +343,83 @@ def add(result):
 
             normalized_tasks.append(task)
 
-        return [Task.create(**task) for task in normalized_tasks]
+        tasks = [Task.create(**task) for task in normalized_tasks]
+        # Pass tasks instead of trying to access self.tasks which would make this recursive
+        tasks = self.fetch_errors_results(tasks)
+        return tasks
+
+    def fetch_errors_results(self, tasks: list) -> list:
+        """ Fetch errors and results for test tasks. Define ADR_CONFIG_PATH and adr.cache for the caching to work"""
```

Comment:
I wouldn't put the second sentence in the documentation of the function, it's not really directly related (otherwise, we'd have to put the same sentence in all functions using adr :P)